packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.md

```diff
@@ -62,7 +62,169 @@ class SimpleApplicationLauncher extends React.Component {
 }
 ```
 
-## Application launcher with tooltip
+## App launcher disabled
+```js
+import React from 'react';
+import { ApplicationLauncher, ApplicationLauncherItem } from '@patternfly/react-core';
+
+class SimpleApplicationLauncher extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isOpen: false
+    };
+    this.onToggle = isOpen => {
+      this.setState({
+        isOpen
+      });
+    };
+    this.onSelect = event => {
+      this.setState({
```

Comment:
I think a better practice is to use the function form of `setState` for simple toggles like this:

```
this.setState( prevState => ({ isOpen: !prevState.isOpen }) );
```

https://medium.com/@wisecobbler/using-a-function-in-setstate-instead-of-an-object-1f5cfd6e55d1

Reply:
Quote from the article:

> I am relying on the value of this.state.showForm to modify the next state of my form. In this simple example, we probably would not run into any issues by relying on this value, but you might imagine that as an application gets more complex and there are multiple setState calls happening and queuing up data to be rendered to the DOM, there is a possibility that the actual value of this.state.showForm might not be what you think.

I think it's fine to use `setState({})` for simple examples.

Reply:
Leaving as-is for now to match existing examples, we may want to refactor all examples if this change does end up being necessary.